plugins/generator-1.20.1/forge-1.20.1/templates/gui/gui_window.java.ftl

```diff
@@ -110,8 +110,20 @@ public class ${name}Screen extends AbstractContainerScreen<${name}Menu> {
 			<#if hasProcedure(component.displayCondition)>
 				if (<@procedureOBJToConditionCode component.displayCondition/>)
 			</#if>
-				if (mouseX > leftPos + ${x} && mouseX < leftPos + ${x + component.width} && mouseY > topPos + ${y} && mouseY < topPos + ${y + component.height})
-					guiGraphics.renderTooltip(font, <#if hasProcedure(component.text)>Component.literal(<@procedureOBJToStringCode component.text/>)<#else>Component.translatable("gui.${modid}.${registryname}.${component.getName()}")</#if>, mouseX, mouseY);
+				if (mouseX > leftPos + ${x} && mouseX < leftPos + ${x + component.width} && mouseY > topPos + ${y} && mouseY < topPos + ${y + component.height}) {
+					<#if hasProcedure(component.text)>
+					String hoverText = <@procedureOBJToStringCode component.text/>;
+					if (hoverText != null) {
+						if (hoverText.contains("\n"))

```

Comment:
Do we need this extra \n check?

Reply:
Methods for single component and list of components don't call one another, when I was changing this file, I made it "choose" one of 2 methods just because of that.

Reply:
Yes but if it does not contain \n, stream will return element with single entry which should work just fine, wouldn't it?

Reply:
Maybe 🤔